odkx-src/injected-interfaces.rst

```diff
@@ -0,0 +1,1305 @@
+  loggingString
+  getActiveUser
+  getProperty
+  propertyId
+  getBaseUrl
+  setSessionVariable
+  elementPath
+  jsonValue
+  getSessionVariable
+  genUUID
+  constructSurveyUri
+  screenPath
+  elementKeyToValueMap
+  doAction
+  dispatchStruct
+  webkit
+  MediaCaptureImageActivity
+  intentObject
+  setData
+  setPackage
+  setType
+  setAction
+  addCategory
+  componentPackage
+  componentActivity
+  setComponent
+  getProperty
+  appName
+  dispatchStruct
+  viewFirstQueuedAction
+  removeFirstQueuedAction
+  closeWindow
+  resultCode
+  keyValueBundle
+  viewFirstQueuedAction
+  refAction
+  jsonValue
+  urlhash
+  uri
+  opendatakit
+  Url
+  resultCodeOfAction
+  newJsDate
+  formId
+  instanceId
+  isoCountry
+  displayCountry
+  localizeTokenField
+  setResult
+  yyyy
+  ddTHH
+  HH
+  sssssssss
+  timestamp
+  Timestamp
+  org
+  reconstructive
+  getTableMetadata
+  tableMetadataCache
+  getTableMetadataRevision
+  putTableMetadata
+  getOdkDataIf
+  getViewData
+  getRoles
+  getDefaultGroup
+  getUsers
+  getAllTableIds
+  whereClause
+  sqlBindParams
+  groupBy
+  orderByElementKey
+  orderByDirection
+  includeKVS
+  arbitraryQuery
+  sqlCommand
+  sqlBindParams
+  getRows
+  getMostRecentRow
+  changeAccessFilterOfRow
+  defaultAccess
+  rowOwner
+  groupReadOnly
+  groupModify
+  groupPrivileged
+  updateRow
+  deleteRow
+  addRow
+  failureCallbackFn
+  addCheckpoint
+  successCallbackFn
+  failureCallbackFn
+  saveCheckpointAsIncomplete
+  columnNameValueMap
+  saveCheckpointAsComplete
+  deleteAllCheckpoints
+  deleteLastCheckpoint
+  createLocalOnlyTableWithColumns
+  columnNameTypeMap
+  deleteLocalOnlyTable
+  insertLocalOnlyRow
+  updateLocalOnlyRows
+  deleteLocalOnlyRows
+  simpleQueryLocalOnlyTables
+  arbitrarySqlQueryLocalOnlyTables
+  queueRequest
+  invokeCallbackFn
+  jsonResult
+  cbId
+  requestMap
+  responseAvailable
+
+
+ODK-X WebKit
+===================================
+
+.. _injected-interfaces:
+
+.. contents:: :local:
+
+The Java framework on the Android device injects two Java interfaces (`odkCommonIf` and `odkDataIf`) into both Tables and Survey WebKits. Additionally, it injects one additional Java interface into each: `odkTablesIf` into Tables WebKits and `odkSurveyStateManagement` into Survey WebKits.
+
+Within the Javascript, it is expected that all interactions with these interfaces will be done through wrapper objects. Specifically, for *odkCommonIf* and *odkDataIf*, Javascript programmers would invoke methods on the `odkCommon` and `odkData` objects defined in
+
+  - :file:`system/js/odkCommon.js`
+  - :file:`system/js/odkData.js`
+
+The Tables-specific interface is interacted with via the `odkTables` object defined in:
+
+  - :file:`system/tables/js/odkTables.js`
+
+This wrapper object mostly invokes `odkCommon` to perform its actions, but does call the `odkTablesIf` injected interface's one method to load the list view portion of the split-screen detail-with-list-view layout.
```

Comment:
I would remove line 193. Simply state that odkTables.js is for controlling the tables app.